### PR TITLE
Leverage str_ends_with

### PR DESCRIPTION
**Pull request by Tobion**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | no 
| Deprecations? | no 
| Tickets       | 
| License       | MIT
| Doc PR        | 

added the php80 polyfill to requirements when necessary. some components already had the requirement anyway.
Related to #41576

### CODE REVIEW

**Review by derrabus — APPROVED** [review]

File: src/Symfony/Component/Form/composer.json [hunk 1]
```diff
@@ -22,6 +22,7 @@
         "symfony/options-resolver": "~4.3|^5.0",
         "symfony/polyfill-ctype": "~1.8",
         "symfony/polyfill-mbstring": "~1.0",
+        "symfony/polyfill-php80": "^1.15",
```
Comment (noniagriconomie) [inline src/Symfony/Component/Form/composer.json]: shouldnt be `1.16` according to https://github.com/symfony/polyfill-php80/commit/cb64c503a89a49bcf6334a57639e244e488b0fe9?

Comment (stof) [reply src/Symfony/Component/Form/composer.json]: it should, indeed

**Comment by fabpot** [comment]
See my comment here: https://github.com/symfony/symfony/pull/41576#issuecomment-874537201

File: src/Symfony/Component/Form/composer.json (hunk 1 above)
Comment (Tobion) [reply src/Symfony/Component/Form/composer.json]: good catch @noniagriconomie. fixed it

**Comment by Tobion** [comment]
Could we merge this to not create conflicts? We don't need to wait for CS fixer to have the relevant rules before merging this. We can just enable the rule when it's available.

**Comment by nicolas-grekas** [comment]
Thank you @Tobion.

**Pull request merged**